src/main/java/teammates/ui/webapi/action/SearchStudentsAction.java

```diff
@@ -0,0 +1,33 @@
+package teammates.ui.webapi.action;
+
+import java.util.List;
+
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Const;
+import teammates.ui.webapi.output.SearchStudentsData;
+
+public class SearchStudentsAction extends Action {
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        // Only instructors and admins can search for student
+        if (userInfo.isStudent) {
+            throw new UnauthorizedAccessException("Instructor or Admin privilege is required to access this resource.");
+        }
+    }
+
+    @Override
+    public ActionResult execute() {
+        String searchKey = getNonNullRequestParamValue(Const.ParamsNames.SEARCH_KEY);
+        List<InstructorAttributes> instructors = logic.getInstructorsForGoogleId(userInfo.id);
+
+        List<StudentAttributes> students = logic.searchStudents(searchKey, instructors).studentList;
```

Comment:
Seems like `logic.searchStudents` only applies to instructors; the current admin search endpoint uses `logic.searchStudentsInWholeSystem`. The difference is that `searchStudents` limits the results to only students in courses that the querying user is an instructor of, whereas `searchStudentsInWholeSystem` searches all students with no conditions. We might need a conditional here to call the right method so that the access permissions are enforced.

Reply:
hmm true dat

Reply:
fixed